### PR TITLE
[TRACKER-282] - Fix when explore is disabled, the preview tab in Tracker keeps spinning.

### PR DESCRIPTION
**Pull request by sushil-car**
https://issues.cask.co/browse/TRACKER-282

### CODE REVIEW

File: cdap-ui/app/tracker/controllers/preview-ctrl.js [hunk 1]
```diff
@@ -38,19 +39,22 @@ class TrackerPreviewController {
...
+    let query = 'SELECT * FROM ' + this.entityType + '_' + this.$state.params.entityId + ' LIMIT 500';
 
     this.myTrackerApi.postQuery(params, { query: query })
       .$promise
       .then((res) => {
         this.pollQueryStatus(res.handle);
+      }, (err) => {
+        this.loading = false;
+        this.previewNotAvailable = true;
+        console.log('Error', err);
```
Comment (ajainarayanan) [inline cdap-ui/app/tracker/controllers/preview-ctrl.js]: We show appropriate message. We can remove this console.log statement.

File: cdap-ui/app/tracker/controllers/preview-ctrl.js [hunk 2]
```diff
@@ -38,19 +39,22 @@ class TrackerPreviewController {
...
     if (this.$state.params.entityType === 'streams') {
-      entityType = 'stream';
+      this.entityType = 'stream';
     } else if (this.$state.params.entityType === 'datasets') {
-      entityType = 'dataset';
+      this.entityType = 'dataset';
     }
 
-    let query = 'SELECT * FROM ' + entityType + '_' + this.$state.params.entityId + ' LIMIT 500';
+    let query = 'SELECT * FROM ' + this.entityType + '_' + this.$state.params.entityId + ' LIMIT 500';
```
Comment (ajainarayanan) [inline cdap-ui/app/tracker/controllers/preview-ctrl.js]: This query is incorrect. We should use the `ExploreTablesStore` to get the database name and table name. Please refer Explore Fast action in the react code base

File: cdap-ui/app/tracker/templates/preview.html [hunk 3]
```diff
@@ -59,3 +59,10 @@
   ng-if="PreviewController.previewData.length === 0 && !PreviewController.loading">
   <p>No data</p>
```
Comment (ajainarayanan) [inline cdap-ui/app/tracker/templates/preview.html:60]: Would this show both `No data` and the `Preview is not available for this ...` message?

**Review by ajainarayanan — CHANGES_REQUESTED** [review]

File: cdap-ui/app/tracker/controllers/preview-ctrl.js (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/app/tracker/controllers/preview-ctrl.js]: `ExploreTablesStore` needs to be exported via `common/cask-header.js` and then used here via `window` object.

File: cdap-ui/app/tracker/templates/preview.html (hunk 3 above)
Comment (sushil-car) [reply cdap-ui/app/tracker/templates/preview.html:60]: Preview message will be shown only when **previewNotAvailable** gets true and when it gets true, this  means error occurred while executing query and hence **previewData** will be undefined and its length won't be equal to zero. 

On the other hand **previewNotAvailable** wont get true if query is successful and even when we have empty data.

File: cdap-ui/app/tracker/controllers/preview-ctrl.js (hunk 2 above)
Comment (sushil-car) [reply cdap-ui/app/tracker/controllers/preview-ctrl.js]: Tried exporting ExploreTablesStore  and calling window.CaskCommon.ExploreTablesStore.getState() but its returning empty array of tables.
Do I also need to import ActionCreater here and call fetchTables method to get the tables list or am I missing something here ?

**Comment by ajainarayanan** [comment]
The change doesn't fix the infinite spinning issue.

**Comment by nitinmotgi** [comment]
What are we doing with this PR ? Close it incomplete then atleast.

**Comment by ajainarayanan** [comment]
Closing this for now as it is of lower priority. Will reopen again with appropriate fix.

**Pull request closed without merging**